File: Chapter04/FS/VerticalOptionsDemo/VerticalOptionsDemo/VerticalOptionsDemo/VerticalOptionsDemoPage.fs
namespace VerticalOptionsDemo

open System
open System.Reflection
open Xamarin.Forms

type VerticalOptionsDemoPage() = 
    inherit ContentPage()

    let stackLayout = StackLayout()
    let colors = [| Color.Yellow; Color.Blue |]

    do typeof<LayoutOptions>.GetRuntimeFields()
        |> Seq.filter (fun field -> field.IsPublic && field.IsStatic)
        |> Seq.sortBy (fun field -> (field.GetValue(null) :?> LayoutOptions).Alignment)
        |> Seq.map (fun field -> Label(Text = "VerticalOptions = " + field.Name,
                                       VerticalOptions = (field.GetValue(null) :?> LayoutOptions),
                                       HorizontalTextAlignment = TextAlignment.Center,
                                       FontSize = Device.GetNamedSize(NamedSize.Medium, typeof<Label>),
                                       TextColor = colors.[stackLayout.Children.Count % 2],
                                       BackgroundColor = colors.[1 - stackLayout.Children.Count % 2]))
        |> Seq.iter (fun label -> stackLayout.Children.Add(label))

    do base.Padding <- match Device.RuntimePlatform with 
                            | Device.iOS -> Thickness(0.0, 20.0, 0.0, 0.0)
                            | _ -> Thickness()

    do base.Content <- stackLayout

File: Chapter04/VerticalOptionsDemo/VerticalOptionsDemo/VerticalOptionsDemo/VerticalOptionsDemoPage.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Reflection;
using Xamarin.Forms;

namespace VerticalOptionsDemo
{
    public class VerticalOptionsDemoPage : ContentPage
    {
        public VerticalOptionsDemoPage()
        {
            Color[] colors = { Color.Yellow, Color.Blue };
            int flipFlopper = 0;

            // Create Labels sorted by LayoutAlignment property.
            IEnumerable<Label> labels =
                from field in typeof(LayoutOptions).GetRuntimeFields()
                where field.IsPublic && field.IsStatic
                orderby ((LayoutOptions)field.GetValue(null)).Alignment
                select new Label
                {
                    Text = "VerticalOptions = " + field.Name,
                    VerticalOptions = (LayoutOptions)field.GetValue(null),
                    HorizontalTextAlignment = TextAlignment.Center,
                    FontSize = Device.GetNamedSize(NamedSize.Medium, typeof(Label)),
                    TextColor = colors[flipFlopper],
                    BackgroundColor = colors[flipFlopper = 1 - flipFlopper]
                };

            // Transfer to StackLayout.
            StackLayout stackLayout = new StackLayout();

            foreach (Label label in labels)
            {
                stackLayout.Children.Add(label);
            }

            Padding = new Thickness(0, Device.RuntimePlatform == Device.iOS ? 20 : 0, 0, 0);
            Content = stackLayout;
        }
    }
}
